exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillRelOptUtil.java

```diff
@@ -224,4 +226,64 @@ public Void visitInputRef(RexInputRef inputRef) {
     }
   }
 
+  public static boolean isLimit0(RexNode fetch) {
+    if (fetch != null && fetch.isA(SqlKind.LITERAL)) {
+      RexLiteral l = (RexLiteral) fetch;
+      switch (l.getTypeName()) {
+        case BIGINT:
+        case INTEGER:
+        case DECIMAL:
+          if (((long) l.getValue2()) == 0) {
+            return true;
+          }
+      }
+    }
+    return false;
+  }
+
+  public static boolean isProjectOutputRowcountUnknown(RelNode project) {
+    assert project instanceof Project : "Rel is NOT an instance of project!";
+    try {
+      RexVisitor<Void> visitor =
```

Comment:
Would FLATTEN ever occur within other expressions ?  I believe it always occurs as an independent expression.  If that's the case, it seems to me that having a visitor is overkill.. what do you think ?  Even the original rewrite from project to flatten just iterates over the project exprs here [1].  

[1]  https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/visitor/RewriteProjectToFlatten.java#L77

Reply:
Yes, you are correct. If the rewrite does not consider it as embedded within other expressions then it is fine for the utility function to do the same.